### PR TITLE
SW-635 Design Store DEV settings page returning an error while closed login loop happens

### PR DESCRIPTION
**Pull request by hungerpirat**
fixes SW-635

replaced 6 occurences of self.loginState.currentUser?.().settings with a helper method self.getUserSettings() - which uses self.loginState.currentUser?.()?.settings <- with ? behind ().

### CODE REVIEW

File: octoprint_mrbeam/static/js/design_store.js [hunk 1]
```diff
@@ -209,7 +223,7 @@ $(function () {
                 error: function (jqXHR, textStatus, errorThrown) {
                     console.error(
                         "Store bought design saving failed with status " +
-                        jqXHR.status,
+                            jqXHR.status,
```
Comment (Yelko-mrbeam) [inline octoprint_mrbeam/static/js/design_store.js:226]: There are some formatting differences. Do we have a standard formatting ruleset to verify which one is the correct one?

File: octoprint_mrbeam/static/js/design_store.js [hunk 2]
```diff
@@ -21,44 +21,43 @@ $(function () {
             self.prepareDesignStoreTab();
         };
 
+        self.getUserSettings = function () {
+            const settings = self.loginState.currentUser?.()?.settings;
+            return settings;
+        };
```
Comment (Yelko-mrbeam) [inline octoprint_mrbeam/static/js/design_store.js:27]: Nice refactoring, much more readable

**Review by Yelko-mrbeam — APPROVED** [review]

File: octoprint_mrbeam/static/js/design_store.js (hunk 1 above)
Comment (khaledsherkawi) [reply octoprint_mrbeam/static/js/design_store.js:226]: Yea I believe we agreed on "prettier" at one point but it's not used by everyone. I believe we can approve this for now as we will be deciding on new formatters very soon among the team.

**Review by khaledsherkawi — APPROVED** [review]

File: octoprint_mrbeam/static/js/design_store.js (hunk 2 above)
Comment (khaledsherkawi) [reply octoprint_mrbeam/static/js/design_store.js:27]: Indeed

**Pull request merged**